src/user-lists/user-list.controller.ts

```diff
@@ -102,4 +107,74 @@ export class UserListController {
 
     await this.userListService.deleteUserList(list.id);
   }
+
+  @Get("/contributors")
+  @ApiOperation({
+    operationId: "getContributors",
+    summary: "Retrieves paginated contributors",
+  })
+  @ApiBearerAuth()
+  @UseGuards(SupabaseGuard)
+  @ApiPaginatedResponse(DbUserListContributor)
+  @ApiOkResponse({ type: DbUserListContributor })
+  @ApiNotFoundResponse({ description: "Unable to get contributors" })
+  @ApiBadRequestResponse({ description: "Invalid request" })
+  async getContributors(@Query() pageOptionsDto: PageOptionsDto): Promise<PageDto<DbUserListContributor>> {
+    return this.userListService.findAllContributors(pageOptionsDto);
+  }
+
+  @Get("/:id/contributors")
+  @ApiOperation({
+    operationId: "getUserListContributors",
+    summary: "Retrieves contributors for an individual user list",
+  })
+  @ApiBearerAuth()
+  @UseGuards(SupabaseGuard)
+  @ApiPaginatedResponse(DbUserListContributor)
+  @ApiOkResponse({ type: DbUserListContributor })
+  @ApiNotFoundResponse({ description: "Unable to get user list contributors" })
+  @ApiBadRequestResponse({ description: "Invalid request" })
+  @ApiParam({ name: "id", type: "string" })
+  async getUserListContributors(
+    @Query() pageOptionsDto: PageOptionsDto,
+    @Param("id") id: string
+  ): Promise<PageDto<DbUserListContributor>> {
+    return this.userListService.findContributorsByListId(pageOptionsDto, id);
+  }
+
+  @Post("/:id/contributors")
+  @ApiOperation({
+    operationId: "postUserListContributors",
+    summary: "Add new contributors to an individual user list",
+  })
+  @ApiBearerAuth()
+  @UseGuards(SupabaseGuard)
+  @ApiNotFoundResponse({ description: "Unable to add to user list contributors" })
+  @ApiBadRequestResponse({ description: "Invalid request" })
+  @ApiParam({ name: "id", type: "string" })
+  async postUserListContributors(
+    @Body() updateCollaboratorsDto: CollaboratorsDto,
+    @Param("id") id: string
+  ): Promise<DbUserListContributor[]> {
+    const contributors = updateCollaboratorsDto.contributors.map(async (contributorId) =>
+      this.userListService.addUserListContributor(id, contributorId)
+    );
+
+    return Promise.all(contributors);
+  }
+
+  @Delete("/:id/contributors")
```

Comment:
We should be more specific here to only delete the individual record. The `:id` is the list ID

```suggestion
  @Delete("/:id/contributors/:userListContributorId")
```